go/vt/vtgateproxy/discovery.go

```diff
@@ -258,8 +276,22 @@ func (b *JSONGateResolverBuilder) parse() (bool, error) {
 			return false, fmt.Errorf("error parsing JSON discovery file %s: port field %s has invalid value %v", b.jsonPath, b.portField, port)
 		}
 
-		targets = append(targets, targetHost{fmt.Sprintf("%s:%s", address, port), poolType.(string), affinity.(string)})
+		target := targetHost{fmt.Sprintf("%s:%s", address, port), poolType.(string), affinity.(string)}
+		targets[target.PoolType] = append(targets[target.PoolType], target)
+	}
+
+	for poolType := range targets {
+		if b.affinityField != "" {
+			sort.Slice(targets[poolType], func(i, j int) bool {
```

Comment:
I think we want to _sort_ here to pull the same-AZ hosts to the front of the list, but I don't think we want to _slice_ out the ones in the other AZ.

If we don't have numConnections hosts in the same AZ, I think we still want the discovery to try to find that many valid targets, even if that means crossing AZs.

Reply:
we're not dropping any targets here, only sorting

Reply:
🤦 Oh duh of course, my bad

Reply:
I should actually get coffee before reviewing code next time.